server/src/queries.rs

```diff
@@ -0,0 +1,252 @@
+    let aircrafts = storage_client
+        .aircrafts(Request::new(AircraftFilter {})) //todo filter associated aircrafts to dep vertiport?
+        .await?
+        .into_inner()
+        .aircrafts;
+    for _aircraft in aircrafts {
+        let aircraft_schedule = Calendar::from_str(SAMPLE_CAL).unwrap(); //TODO get from aircraft.schedule
+        let is_aircraft_available =
+            aircraft_schedule.is_available_between(departure_time, arrival_time);
+        let aircraft_flights: Vec<FlightPlan> = vec![];
+        /*todo storage_client
+        .flight_plans(Request::new(FlightPlanFilter {}))
+        //todo filter flight_plans(estimated_departure_between: ($from, $to), estimated_arrival_between: ($from2, $to2) aircraft_id: $ID)
+        .await?
+        .into_inner()
+        .flight_plans;*/
+        if !is_aircraft_available || !aircraft_flights.is_empty() {
+            return Err(Status::not_found("Aircraft not available"));
+        }
+    }
+    //6. TODO: check other constraints (cargo weight, number of passenger seats)
+    //7. create draft flight plan (in memory)
+    let fp_id = Uuid::new_v4().to_string();
+    unconfirmed_flight_plans().lock().unwrap().insert(
+        fp_id.clone(),
+        FlightPlan {
+            id: 1234, //todo string id
+            flight_status: FlightStatus::Ready as i32,
+        },
+    );
+    //8. automatically cancel draft flight plan if not confirmed by user
+    cancel_flight_after_timeout(fp_id.clone());
+    //9. return response - TODO copy from storage flight plan
+    let item = QueryFlightPlan {
+        id: fp_id,
+        pilot_id: 1,
+        aircraft_id: 1,
+        cargo: [123].to_vec(),
+        weather_conditions: "Sunny, no wind :)".to_string(),
+        vertiport_id_departure: 1,
+        pad_id_departure: 1,
+        vertiport_id_destination: 1,
+        pad_id_destination: 1,
+        estimated_departure: Some(Timestamp {
+            seconds: departure_time.timestamp(),
+            nanos: departure_time.timestamp_subsec_nanos() as i32,
+        }),
+        estimated_arrival: Some(Timestamp {
+            seconds: arrival_time.timestamp(),
+            nanos: arrival_time.timestamp_subsec_nanos() as i32,
+        }),
+        actual_departure: None,
+        actual_arrival: None,
+        flight_release_approval: None,
+        flight_plan_submitted: None,
+        flight_status: FlightStatus::Ready as i32,
+        flight_priority: FlightPriority::Low as i32,
+    };
+    let response = QueryFlightResponse {
+        flights: [item].to_vec(),
+    };
+    Ok(Response::new(response))
+}
+
+fn get_fp_by_id(id: String) -> Option<FlightPlan> {
+    unconfirmed_flight_plans().lock().unwrap().get(&id).copied()
+}
+
+///Confirms the flight plan
+pub async fn confirm_flight(
+    request: Request<Id>,
+    mut storage_client: StorageRpcClient<tonic::transport::Channel>,
+) -> Result<Response<ConfirmFlightResponse>, Status> {
+    let fp_id = request.into_inner().id;
```

Comment:
Check if valid UUID format before the get_fp_by_id operation

Reply:
I suppose if it gets rejected then one of the possible explanations is that it's not in proper UUID format

We'll improve on the error messages in a future PR